### PR TITLE
PS-8564 8.0.31-23 release notes and release note tasks (8.0)

### PR DESCRIPTION
**Pull request by alina-derkach-oaza**
new file:   docs/release-notes/8.0.31-23.md
modified:   docs/release-notes/release-notes_index.md
modified:   mkdocs-base.yml
modified:   copyright-and-licensing-information.md

### CODE REVIEW

File: docs/release-notes/8.0.31-23.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+Improvements and bug fixes introduced by Oracle for *MySQL* 8.0.31 and included in *Percona Server for MySQL* are the following:
+
+* Now a replica can add a Generated Invisible Primary Keys(GIPK) to any InnoDB table. To achieve this behavior, the `GENERATE` value is added as a possible value for the `CHANGE REPLICATION SOURCE TO` statement's `REQUIRE_TABLE_PRIMARY_KEY_CHECK` option.
+
+    `REQUIRE_TABLE_PRIMARY_KEY_CHECK = GENERATE` option can be used on a per-channel basis.
+
+    !!! note
+
+        A replica ignores any settings for `sql_generate_invisible_primary_key` if these settings are made on the source, and this variable is not replicated. This behavior is inherited from the previous releases.
```
Comment (patrickbirch) [inline docs/release-notes/8.0.31-23.md]: No reason for this note. Just add the information to the main text.

File: docs/release-notes/8.0.31-23.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+* `innodb_encryption_rotation_loops`
+
+* `innodb_default_encryption_key_id`
+
+* `rotate_system_key` and any dependencies
+
+This release changes the following system variable options:
+
+* `ONLINE_TO_KEYRING` and `ONLINE_FROM_KEYRING_TO_UNENCRYPTED` options have been removed in the `default_table_encryption` variable.
```
Comment (patrickbirch) [inline docs/release-notes/8.0.31-23.md]: options are removed.
Not past tense.

File: docs/release-notes/8.0.31-23.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+* `innodb_default_encryption_key_id`
+
+* `rotate_system_key` and any dependencies
+
+This release changes the following system variable options:
+
+* `ONLINE_TO_KEYRING` and `ONLINE_FROM_KEYRING_TO_UNENCRYPTED` options have been removed in the `default_table_encryption` variable. 
+
+* The `innodb_sys_tablespace_encrypt` variable data type has been changed to `Boolean`.
```
Comment (patrickbirch) [inline docs/release-notes/8.0.31-23.md]: is changed
not past tense

File: docs/release-notes/8.0.31-23.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+* [PS-8433](https://jira.percona.com/browse/PS-8433): Reduces memory spent when caching triggers by replacing a fixed-size error buffer with a variable-size error buffer.
+
+## Bug fixes
+
+* [PS-1098](https://jira.percona.com/browse/PS-1098): Manually rotating the log files by calling `audit_log_flush` multiple times resulted in using a wrong log file.
+
+* [PS-8106](https://jira.percona.com/browse/PS-8106): MySQL exited when the replication was stopped on replicating a CTAS.
+
+* [PS-8128](https://jira.percona.com/browse/PS-8128): In-place upgrade of views that refer to `performance_schema` tables failed.
```
Comment (patrickbirch) [inline docs/release-notes/8.0.31-23.md]: An in-place upgrade

File: docs/release-notes/8.0.31-23.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+* [PS-7963](https://jira.percona.com/browse/PS-7963): The trigger definitions are not cached in the table cache when using SELECT statements.
+
+* [PS-8433](https://jira.percona.com/browse/PS-8433): Reduces memory spent when caching triggers by replacing a fixed-size error buffer with a variable-size error buffer.
+
+## Bug fixes
+
+* [PS-1098](https://jira.percona.com/browse/PS-1098): Manually rotating the log files by calling `audit_log_flush` multiple times resulted in using a wrong log file.
+
+* [PS-8106](https://jira.percona.com/browse/PS-8106): MySQL exited when the replication was stopped on replicating a CTAS.
```
Comment (patrickbirch) [inline docs/release-notes/8.0.31-23.md]: on replicating a `CREATE TABLE AS` statement.

Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md]: Updated

File: docs/release-notes/8.0.31-23.md (hunk 4 above)
Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md]: Updated

File: docs/release-notes/8.0.31-23.md (hunk 3 above)
Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md]: Updated

File: docs/release-notes/8.0.31-23.md (hunk 2 above)
Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md]: Updated

File: docs/release-notes/8.0.31-23.md (hunk 1 above)
Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md]: Added to the main text.

File: docs/release-notes/8.0.31-23.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+* Now a replica can add a Generated Invisible Primary Keys(GIPK) to any InnoDB table. To achieve this behavior, the `GENERATE` value is added as a possible value for the `CHANGE REPLICATION SOURCE TO` statement's `REQUIRE_TABLE_PRIMARY_KEY_CHECK` option.
+
+    `REQUIRE_TABLE_PRIMARY_KEY_CHECK = GENERATE` option can be used on a per-channel basis.
+
+    Setting `sql_generate_invisible_primary_key` on the source is ignored by a replica because this variable is not replicated. This behavior is inherited from the previous releases.
+
+* MySQL adds support for the SQL standard [`INTERSECT`](https://dev.mysql.com/doc/refman/8.0/en/intersect.html) and [`EXCEPT`](https://dev.mysql.com/doc/refman/8.0/en/except.html) table operators.
+
+* Now, InnoDB supports parallel index builds. This improves index build performance. The sorted index entries are loaded into a B-tree in a multithread. In previous releases, this action was performed by a single thread.
```
Comment (patrickbirch) [inline docs/release-notes/8.0.31-23.md]: Now

File: docs/release-notes/8.0.31-23.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+
+* Now, the Performance and sys schemas show metrics for the global and session memory limits introduced in MySQL 8.0.28.
```
Comment (patrickbirch) [inline docs/release-notes/8.0.31-23.md]: Now

File: docs/release-notes/8.0.31-23.md (hunk 6 above)
Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md]: removed

File: docs/release-notes/8.0.31-23.md (hunk 7 above)
Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md]: removed

File: docs/release-notes/8.0.31-23.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+* [PS-8128](https://jira.percona.com/browse/PS-8128): An in-place upgrade of views that refer to `performance_schema` tables failed.
+
+* [PS-8328](https://jira.percona.com/browse/PS-8328): MySQL 8.0 exited when using `group_concat` with `group by with rollup`.
+
+* [PS-8503](https://jira.percona.com/browse/PS-8503): `checkpoint_age` related status had no value. (Thanks to 
+Huang Zhiyong for reporting this issue.)
+
+* [PS-8547](https://jira.percona.com/browse/PS-8547): Release build crashed with Kmip. (Thanks to user Elbandi for reporting this issue.)
+
+## Deprecated and removed
```
Comment (patrickbirch) [inline docs/release-notes/8.0.31-23.md:102]: Can we bold this title to match the other titles? It tends to blend into the text.

Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md:102]: All the titles have the same format

File: docs/release-notes/8.0.31-23.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+    WHERE NAME='memory/sql/NET::buff';
+    ```
+
+* The `audit_log_flush` variable has been deprecated and will be removed in future releases.
+
+Find the full list of bug fixes and changes in the [MySQL 8.0.31 Release Notes](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-31.html).
+
+## Improvements
+
+* [PS-7963](https://jira.percona.com/browse/PS-7963): The trigger definitions are not cached in the table cache when using SELECT statements.
```
Comment (percona-ysorokin) [inline docs/release-notes/8.0.31-23.md]: @alina-derkach-oaza May be we should give a link to the new `table_open_cache_triggers` variable that this improvement introduces.

Comment (percona-ysorokin) [reply docs/release-notes/8.0.31-23.md]: as well as
```
table_open_cache_triggers_hits
table_open_cache_triggers_misses
table_open_cache_triggers_overflows
```
status variables

Comment (alina-derkach-oaza) [reply docs/release-notes/8.0.31-23.md]: updated

**Review by patrickbirch — APPROVED** [review]
Good

File: mkdocs-base.yml [hunk 10]
```diff
@@ -185,31 +199,34 @@ nav:
...
       - security/data-masking.md
-      - security/system-variables.md
+      - LDAP features:
+        - security/ldap-authentication.md
+        - security/ldap-system-variables.md
+      - Data-at-rest-encryption:
+        - security/data-at-rest-encryption.md
+        - security/using-keyring-plugin.md
+        - Authentication:
+          - security/using-kmip.md
```
Comment (percona-ysorokin) [inline mkdocs-base.yml]: strange indentation

File: variables.yml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+# variables for HTML output
+
+release: '8.0.31-23'
+version: '8.0'
+release-date: 2023
```
Comment (percona-ysorokin) [inline variables.yml]: no new line at EOF

File: mkdocs-base.yml (hunk 10 above)
Comment (alina-derkach-oaza) [reply mkdocs-base.yml]: This is required to have the third level list in the side table of contents. It works.

File: variables.yml (hunk 11 above)
Comment (alina-derkach-oaza) [reply variables.yml]: Added

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**